### PR TITLE
First pass at social tooltip with attribution

### PR DESCRIPTION
**Pull request by cvializ**
@andrewconner @JRuff42 @eishay 
uses the existing tracking mechanisms
screen shot 2016-03-01 at 5 34 29 pm

### CODE REVIEW

File: packrat/scripts/keeper.js [hunk 1]
```diff
@@ -509,36 +509,65 @@ k.keeper = k.keeper || function () {  // idempotent for Chrome
...
+    };
+  }
+
+  var slackUserEntityRe = /<@U.*?>/g;
+  function massageSlackEntities(message) {
+    var matches = [];
+    message = message.replace(slackUserEntityRe, '');
+
+    // Fill an array with match objects from slackUrlEntityRe
+    for (var m, slackUrlEntityRe = /<(.*?)\|(.*?)>|<(.*?)>/g; m = slackUrlEntityRe.exec(message); matches.push(m)) {}
```
Comment (andrewconner) [inline packrat/scripts/keeper.js:540]: Hah, abusing `for` syntax :100:

File: packrat/scripts/keeper.js [hunk 2]
```diff
@@ -509,36 +509,65 @@ k.keeper = k.keeper || function () {  // idempotent for Chrome
...
+  function getTeamName() {
+    var subdomainRe = /https?:\/\/?(?:([^.]+)\.)?slack\.com/;
+    return function (templateText, render) {
+      var url = getPlainText(render(templateText));
+      var matches = subdomainRe.exec(url);
+      return matches ? matches[1] : '';
+    };
+  }
+
+  var slackUserEntityRe = /<@U.*?>/g;
```
Comment (andrewconner) [inline packrat/scripts/keeper.js]: FYI, I think there's several other entity formats as well, like #channel and groups. I'll dig. Less important to fix immediately.

File: packrat/scripts/keeper.js (hunk 1 above)
Comment (cvializ) [inline packrat/scripts/keeper.js:540]: I'm a little proud of and disgusted by this one liner, not gonna lie haha

**Comment by cvializ** [comment]
got Firefox fixed :ship: :ship: :ship:

**Pull request merged**